src/Parlot/Scanner.cs

```diff
@@ -472,5 +472,56 @@ private bool ReadQuotedString(char quoteChar, out TokenResult result)
 
             return true;
         }
+
+        /// <summary>
+        /// Reads a sequence token enclosed in arbritrary start and end characters.
+        /// </summary>
+        /// <remarks>
+        /// This method doesn't escape the string, but only validates its content is syntactically correct.
+        /// The resulting Span contains the original quotes.
+        /// </remarks>
+        public bool ReadNonEscapableSequence(char startSequenceChar, char endSequenceChar, out TokenResult result)
+        {
+            var startChar = Cursor.Current;
+
+            if (startChar != startSequenceChar)
+            {
+                result = TokenResult.Fail();
+                return false;
+            }
+
+            // Fast path if there aren't any escape char until next quote
+            var startOffset = Cursor.Offset + 1;
+            var lastQuote = startOffset;
+
+            int nextQuote ;
+            do
+            {
+                nextQuote = Cursor.Buffer.IndexOf(endSequenceChar, lastQuote + 1);
+
+                if (nextQuote == -1)
+                {
+                    if(startOffset == lastQuote)
+                    {
+                        // There is no end sequence character, not a valid escapable sequence
+                        result = TokenResult.Fail();
+                        return false;
+                    }
+                    nextQuote = lastQuote - 1;
+                    break;
+                }
+
+                lastQuote = nextQuote + 1;
+            }
+            while(Cursor.Buffer.Length > lastQuote && Cursor.Buffer[lastQuote] == endSequenceChar);
```

Comment:
Infinite loop, or if not then it should be a single `if`. I don't see any state changing in the loop.

Reply:
the purpose is to catch multiple cases, so it has to be a while. It would be an infinite loop if I did not have a break on line 511.

Reply:
There is a ; at the end of the while line. No?

Reply:
it's a do/while, not a while/do ;)